lib/using_restricted_array.rb

```diff
@@ -6,87 +6,144 @@
-# Returns true if found, false otherwise.
-# Time complexity: ?
-# Space complexity: ?
-def search(array, length, value_to_find)
-  raise NotImplementedError
-end
-
-# Finds and returns the largest integer value the array
-# Assumes that the array is not sorted.
-# Time complexity: ?
-# Space complexity: ?
-def find_largest(array, length)
-  raise NotImplementedError
-end
-
-# Finds and returns the smallest integer value in the array
-# Assumes that the array is not sorted.
-# Time complexity: ?
-# Space complexity: ?
-def find_smallest(array, length)
-  raise NotImplementedError
-end
-
-# Reverses the values in the integer array in place
-# Time complexity: ?
-# Space complexity: ?
-def reverse(array, length)
-  raise NotImplementedError
-end
-
-# For an array sorted in ascending order, searches for 'value_to_find'.
-# Returns true if found, false otherwise.
-# Time complexity: ?
-# Space complexity: ?
-def binary_search(array, length, value_to_find)
-  raise NotImplementedError
-end
-
-# Helper method provided to sort the array in ascending order
-# Implements selection sort
-# Time complexity = O(n^2), where n is the number of elements in the array.
-# To find the correct value for index 0, a total of (n-1) comparisons are needed.
-# To find the correct value for index 1, a total of (n-2) comparisons are needed.
-# To find the correct value for index 2, a total of (n-3) comparisons are needed.
-# and so on ...
-# To find the correct value for index (n-2), a total of 1 comparisons is needed.
-# To find the correct value for the last index, a total of 0 comparisons are needed.
-# Total number of comparisons = (n-1) + (n-2) + ... 3 + 2 + 1
-#                             = (n * (n-1))/2
-# This is O(n^2) in Big O terms.
-# Space complexity = constant or O(1) since the additional storage needed,
-# does not depend on input array size.
-def sort(array, length)
-  length.times do |index| # outer loop - n elements
-    min_index = index # assume index is where the next minimally value is
-    temp_index = index+1 # compare with values at index+1 to length-1
-    while temp_index < length # inner loop - n-1 elements
-      if array[temp_index] < array[min_index] # found a new minimum, update min_index
-        min_index = temp_index
+  i = 0
+  while array[i] != nil
+    puts "#{array[i]}"
+    i += 1
+  end
+  
+  # For an unsorted array, searches for 'value_to_find'.
+  # Returns true if found, false otherwise.
+  # Time complexity: OofN linear bc the loop only goes the length
+  # Space complexity: Oof1 constant bc it onky returns true or false
+  def search(array, length, value_to_find)
+    length.times do |i|
+      if array[i] == value_to_find
+        return true
       end
-      temp_index += 1 # move to next index
-    end
-    if min_index != index # next minimum value is not at current index, swap
-      temp = array[min_index]
-      array[min_index] = array[index]
-      array[index] = temp
+    else i += 1
```

Comment:
1.  Your indentation needs work.  Maybe try using the **Rufo** extension to help with code formatting.
2.  You are missing an end
3.  The `.times` loop will increment `i` so you don't need to do that
4.  You should return `false` if you reach the end of the loop and don't find the item you are searching for.